pkg/config/spec.go

```diff
@@ -367,7 +369,7 @@ func setUpgradeSourceSize(cfg *Config, spec *v1.UpgradeSpec) error {
 		targetSpec = &(spec.Active)
 	}
 
-	if targetSpec.Source.IsEmpty() {
+	if targetSpec.Source != nil && targetSpec.Source.IsEmpty() {
```

Comment:
this is an extra check nice to add if you hack around this and suddenly pass an empty source so it doesnt fail as hard :D